packages/beacon-node/src/network/reqresp/beaconBlocksMaybeBlobsByRoot.ts

```diff
@@ -77,26 +82,62 @@ export async function unavailableBeaconBlobsByRoot(
   }
 
   // resolve missing blobs
-  const blobIdentifiers: deneb.BlobIdentifier[] = [];
+  const blobIdentifiersWithCommitments: (deneb.BlobIdentifier & {kzgCommitment: deneb.KZGCommitment})[] = [];
   const slot = block.message.slot;
+  const fork = config.getForkName(slot);
   const blockRoot = config.getForkTypes(slot).BeaconBlock.hashTreeRoot(block.message);
 
   const blobKzgCommitmentsLen = (block.message.body as deneb.BeaconBlockBody).blobKzgCommitments.length;
   for (let index = 0; index < blobKzgCommitmentsLen; index++) {
-    if (blobsCache.has(index) === false) blobIdentifiers.push({blockRoot, index});
+    if (blobsCache.has(index) === false) {
+      blobIdentifiersWithCommitments.push({
+        blockRoot,
+        index,
+        kzgCommitment: (block.message.body as deneb.BeaconBlockBody).blobKzgCommitments[index],
+      });
+    }
   }
 
-  let allBlobSidecars: deneb.BlobSidecar[];
-  if (blobIdentifiers.length > 0) {
-    allBlobSidecars = await network.sendBlobSidecarsByRoot(peerId, blobIdentifiers);
+  const networkReqIdentifiers: deneb.BlobIdentifier[] = [];
+  if (executionEngine !== undefined) {
+    const signedBlockHeader = signedBlockToSignedHeader(config, block);
+    const versionedHashes = blobIdentifiersWithCommitments.map((bi) => kzgCommitmentToVersionedHash(bi.kzgCommitment));
+    const blobAndProofs = await executionEngine
```

Comment:
`executionEngine.getBlobs()` should not be called inside this function because the consumer of this function (which is `unknownBlock`) has a loop over shuffled n peers, it would means that this function is called n times

Reply:
we can keep a flag that it has been called before but the main problem is that when we don't even know the block, we can't call this since we dont know versioned hashes, and we need to call this before start req/resp of blobs from the network

so the best way would be to just track it via  a flag in blockinput